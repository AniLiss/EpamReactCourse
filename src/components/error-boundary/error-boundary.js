import React from 'react';

export class ErrorBoundary extends React.Component {
    static getDerivedStateFromError(error) {
        return {error};
    }

    state = {error: null};

    componentDidCatch(error, info) {
        logger.error(error, info);
    }

    render() {
        if (this.state.error) {
            return (
                <div>
                    Unhandled error in component: {this.state.error.toString()};
                </div>
            );
        }

        return this.props.children;
    }
}
