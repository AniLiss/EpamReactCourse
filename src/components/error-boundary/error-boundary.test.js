import React from 'react';
import {shallow} from 'enzyme';
import {ErrorBoundary} from './error-boundary';

describe('Error Boundary', () => {
    it('is expected to render erroe message', () => {
        const wrapper = shallow(<ErrorBoundary />);
        wrapper.setState({error: 'Something went terribly wrong'});
        expect(wrapper).toMatchSnapshot();
    })
});
