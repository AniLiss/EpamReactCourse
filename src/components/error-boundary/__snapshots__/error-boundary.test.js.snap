// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Boundary is expected to render erroe message 1`] = `
<div>
  Unhandled error in component: 
  Something went terribly wrong
  ;
</div>
`;
